### DISCUSSION TITLE
What guides and projects would you like to see in the docs?

### DISCUSSION THREAD

**Discussion by HipsterBrown** [HipsterBrown/xs-dev#129] [Q&A]
The [initial getting started guide](https://xs-dev.js.org/guide/00-prepare/) is inspired by the old [Tessel Start guide](http://tessel.github.io/t2-start/) to provide a smooth onboarding into using the Moddable SDK for the first time.

The guide aims to cover:
- gpio (analog, digital, pwm)
- connecting to wifi
- connecting to bluetooth
- creating a wifi network (access point)
- creating a bluetooth device
- starting a web server
- interacting with HTTP APIs (send a toot?)

What other topics would you like to see covered in the docs?

What kind of projects would you like to see documented as tutorials, i.e. an [on-air light](https://github.com/HipsterBrown/on-air-light), that brings together the topics covered in the starting guide?

**Comment by bartmichu** [comment]
Hello.

MQTT client is a popular topic and a tutorial on that would be beneficial.

Maybe a gentle introduction to bitwise operators, explaining their purpose and demonstrating some practical use cases.

**Reply by dashcraft** [reply]
We do a bit with MQTT now, although the moddable docs are really, _really_ good, i can potentially help create a small example for xs-dev?

**Reply by HipsterBrown** [reply]
@dashcraft that would be great! Providing a nice entry point to the concept via the xs-dev docs can lead to the more expansive Moddable documentation. 

I think the core pain point with the Moddable docs is discoverability. I'd like to work with the team on hosting their comprehensive content and examples with more focused search and learning paths than GitHub repo search provides today, even just on GitHub Pages.

**Reply by phoddie** [reply]
Funny you should mention it. We just recently started hosting the [Moddable SDK docs on our web site](https://www.moddable.com/documentation/readme) too. Much faster to browse than GitHub. There's absolutely more work to do on helping people navigate these docs, but I thought I'd pass this along.

**Reply by HipsterBrown** [reply]
> We just recently started hosting the Moddable SDK docs on our web site too.

Oh excellent! It looks great. 👏

Is this stable enough for me to link from the xs-dev docs? Is there a way to contribute local search and tagging?

**Reply by phoddie** [reply]
> Oh excellent! It looks great. 👏

Thank you! Hoping it helps developers discover more functionality more easily.

> Is this stable enough for me to link from the xs-dev docs?

Yes!

> Is there a way to contribute local search and tagging?

Both would be welcome. We generate the site from an ad-hoc tool that transforms the markdown through the usual GitHub service. Not sure how best to proceed there, but happy to discuss.

**Comment by bartmichu** [comment]
One more thing comes to my mind. In my case, a lot of frustration came from unstable buttons firing events like crazy - turns out, very easy to deal with programmatically. I think it would be good to mention debouncing or throttling if you do something with push buttons and switches.

**Reply by HipsterBrown** [reply]
Good callout. Adding different ways of handling inputs through debouncing or interrupts or observable streams.
